.github/workflows/merge.yaml

```diff
@@ -0,0 +1,69 @@
+name: Merge to release
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Check for relevant changes
+      uses: dorny/paths-filter@v2
+      id: release
+      with:
+        list-files: shell
+        filters: |
+          updated:
+            - added|modified: [ '*.csl', '*.xml' ]
+          deleted:
+            - deleted: [ '*.csl', '*.xml' ]
+
+    - name: Changed files
+      run: |
+        echo updated: ${{ steps.release.outputs.updated_files }}
+        echo deleted: ${{ steps.release.outputs.deleted_files }}
+
+    - name: Checkout release branch
+      uses: actions/checkout@v2
+      if: steps.release.outputs.updated == 'true' || steps.release.outputs.deleted == true
+      with:
+        ref: 'v1.0.1'
```

Comment:
@retorquere, is this the value we need to (manually) change when we want to start push styles to a different branch (e.g. "v1.0.2"). And do I understand correctly that we should first create a branch by hand before directing the GitHub action to it?

Reply:
Yes, and yes. Since it is my understanding that the release branch should structurally omit some files, I can't just auto-create it and push there, but don't know how you want to set it up.

Reply:
> Since it is my understanding that the release branch should structurally omit some files, I can't just auto-create it and push there, but don't know how you want to set it up.

I think we just want to keep all CSL/XML, README.md, and eventually a LICENSE file once we switch to CC BY-SA 4.0 (https://github.com/citation-style-language/schema/issues/195). Looks like we can just white-list extra files to retain at https://github.com/citation-style-language/styles/pull/5111/files#diff-f9d2712bfa1f66ac85ef91c25d5679746ff2daa618e817b7a648dff40e2412e5R16 ?

Reply:
That is indeed correct -- it is a whitelist for what gets pushed through to release, and even then Sheldon will only push csl files to a styles release, and xml files to a locales release. This workflow could in that sense be made more specific by setting an only-csl or only-xml whitelist, but right now this workflow could be dropped into the locales repo without change.

Reply:
Additionally -- it will *only* overwrite the files changed in the commit, so if you set up an empty branch, this workflow won't automatically populate it.

Reply:
> Additionally -- it will _only_ overwrite the files changed in the commit, so if you set up an empty branch, this workflow won't automatically populate it.

Is that easy to change for initially populate the branch? Otherwise we'd have to set up a workaround for populating the branch with styles/locales with modified timestamps.

Reply:
And another case would be situations where GitHub Actions might be broken, which could lead the branches out of sync if we still make commits. Not sure that is a likely scenario.

Reply:
> Is that easy to change for initially populate the branch? Otherwise we'd have to set up a workaround for populating the branch with styles/locales with modified timestamps.

It's not easy to change, but why not just manually do a checkin of all styles once?

> And another case would be situations where GitHub Actions might be broken, which could cause the branches to get out of sync if we still make commits to "master". Not sure that is a likely scenario, since our CI tests would presumably be broken too.

I suppose a scheduled workflow could do a nightly compare-and-repair. But Sheldon previously didn't do any of this, right?

Reply:
> Actually... not so hard to change, but the [change I pushed to my repo](https://github.com/retorquere/styles/commit/432c92fdaef34358b209df9017c38ba8c802525a) is not showing up in this pull request.

It shows up below. I think we'd only need this whenever we populate a new branch ("v1.0.1" once we start using this workflow, and "v1.0.2." when 1.0.2 is released, etc.), and very occasionally after that. Manual triggering it should be fine.

Thanks! Let us know when this is good to merge.

Reply:
Do keep in mind that if population is needed for anything else than initially populating a new branch, this "repair" would not update the timestamps of the files brought in, as Sheldon only changes the timestamp on the released files. It's possible to update the timestamps on `master` instead, but then you really do not want to use the `csl-bot` account, as you'd have a very hard to prevent commit loop.